nifi-nar-bundles/nifi-site-to-site-reporting-bundle/nifi-site-to-site-reporting-task/src/main/java/org/apache/nifi/reporting/AbstractSiteToSiteReportingTask.java

```diff
@@ -119,8 +120,15 @@
     }
 
     @OnScheduled
-    public void setup(final ConfigurationContext context) throws IOException {
-        siteToSiteClient = SiteToSiteUtils.getClient(context, getLogger());
+    public void setup(final ConfigurationContext configContext) throws IOException {
+        this.configContext = configContext;
```

Comment:
`ReportingContext` has the same access to the properties as `ConfigurationContext`, so there's no need to save this off since you're moving the `SiteToSiteUtils.getClient()` call to `onTrigger()`. You should just be able to change the argument in `getClient` from `ConfigurationContext` to `ReportingContext`.

Reply:
I've done with changes you proposed. Thank you!